source/pdk/1.0/pdk_get_started.md

```diff
@@ -0,0 +1,260 @@
+
+PDK creates the new class manifest and a test template file. You can then write tests in this template to validate your class's behavior.
+
+{:.concept}
+### Specifying class parameters
+
+If your new class should take parameters, specify them on the command line when you generate your class.
+
+Specify the values that parameter accepts, and optionally, specify the data type of the parameter with with the parameter name. You can provide any number of parameters on the command line.
+
+
+For example, to create a new class and define an `ensure` parameter for the class, run:
+
+``` bash
+pdk new class class_name "ensure:Enum['absent','present']"
+```
+
+This command creates a file in `module_name/manifests` named `class_name.pp`, with the ensure parameter defined. It also creates a test file in `module_name/spec/class` named `class_name_spec.rb`. This test file includes a basic template for writing your own unit tests.
+
+
+
+Related topics:
+
+* [Data types](TODO: LINK)
+
+{:.task}
+## Generate a new class
+
+To generate a new class in your module, use the `pdk new class` command.
+
+1. From the command line, in your module's directory, run:
+
+   ``` bash
+   pdk new class class_name "ensure:Enum['absent','present']"
+   ```
+
+{:.concept}
+## Testing your module with PDK
+
+PDK provides tools to help you run unit tests on your module and validate your new module's metadata, syntax, and style.
+
+By default, the PDK module template includes tools that can:
+
+* Validate the `metadata.json`.
+* Validate Puppet syntax.
+* Validate Puppet code style.
+* Validate Ruby code style.
+* Run unit tests.
+
+### Validating modules
+
+The validations included in PDK provide a basic check of the well-formedness of the module and syntax and style of the module's files. You do not need to write any tests for this validation.
+
+By default, the `pdk validate` command validates metadata, Ruby, and Puppet code syntax and style. To run a specific validation, add the name as an argument.
+
+For example, to validate the module's metadata, run `pdk validate metadata`.
+
+To send module validation output to a file, use the `pdk validate` command with the option `--format=format[:target]`.
+
+This option specifies the output format and an output target file. For example, to create a report file `report.xml` in the JUnit format, run `pdk validate --format=junit:report.xml`.
+
+You can specify multiple `--format` options, as long as they all have distinct output targets.
+
+To run validations on a specific directory or file, pass the name of the file or directory as an argument with `pdk validate`.
+
+For example, to run all validations on the `/lib` directory only, run `pdk validate all lib/`. [TODO: what does the `all` do? Wouldn't `pdk validate lib/` run all of the validations?]
```

Comment:
I think the answer to this will be contingent on the outcome of [SDK-295](https://tickets.puppetlabs.com/browse/SDK-295). Currently `pdk validate lib/` works, while `pdk validate all lib/` does not.